### PR TITLE
Ddf 1556: changed SSL/TLS configuration to system properties 

### PR DESCRIPTION
**Pull request by ricklarsen**
In SecuritySettingsService

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/253)

### CODE REVIEW

**Comment by ricklarsen** [comment]
Please review: 
@stustison @ahoffer @djblue @vinamartin @clockard @shaundmorris

File: platform/security/platform-security-settings/src/main/java/ddf/security/settings/impl/SecuritySettingsServiceImpl.java [hunk 1]
```diff
@@ -21,6 +21,8 @@
 import java.security.NoSuchAlgorithmException;
```
Comment (stustison) [inline platform/security/platform-security-settings/src/main/java/ddf/security/settings/impl/SecuritySettingsServiceImpl.java]: This entire class should be removed.

**Comment by stustison** [comment]
Check WebSSOTokenValidator and AbstractStsRealm as well.

**Comment by stustison** [comment]
Should be able to also remove CommonSSLFactory.

**Comment by vinamartin** [comment]
:+1:

**Comment by clockard** [comment]
Probably want to clean up commits since it looks like some of them comment out stuff and then uncomment the same code in a later commit.

**Comment by ricklarsen** [comment]
Added new commits, please re-review: 
@clockard @djblue @vinamartin @stustison @shaundmorris

**Comment by clockard** [comment]
:+1:

**Comment by roelens8** [comment]
:+1:

**Comment by spearskw** [comment]
<thumbs-up>

File: catalog/spatial/ogc/spatial-ogc-common/src/test/java/org/codice/ddf/spatial/ogc/catalog/common/TestTrustedRemoteSource.java [hunk 2]
```diff
@@ -163,6 +145,16 @@ public static void tearDownAfterClass() {
     /**
      * Tests that server properly accepts trusted certificates.
      */
+//    @Test
```
Comment (shaundmorris) [inline catalog/spatial/ogc/spatial-ogc-common/src/test/java/org/codice/ddf/spatial/ogc/catalog/common/TestTrustedRemoteSource.java]: @ricklarsen did you mean to just remove this commented out test? Looks like you have it below as well.

File: catalog/spatial/ogc/spatial-ogc-common/src/test/java/org/codice/ddf/spatial/ogc/catalog/common/TestTrustedRemoteSource.java [hunk 3]
```diff
@@ -172,44 +164,93 @@ public void testGoodCertificates() {
...
-            if (remoteSource.get() != null) {
-                fail("Server should have errored out with bad certificate but request passed instead.");
-            }
-        } catch (Exception e) {
-            assertThat(e.getCause(),
-                    anyOf(is(SSLHandshakeException.class), is(SocketException.class)));
-        }
-
-    }
+//    @Test
```
Comment (shaundmorris) [inline catalog/spatial/ogc/spatial-ogc-common/src/test/java/org/codice/ddf/spatial/ogc/catalog/common/TestTrustedRemoteSource.java]: Should all these commented out tests be removed?

File: catalog/spatial/ogc/spatial-ogc-common/src/test/java/org/codice/ddf/spatial/ogc/catalog/common/TestTrustedRemoteSource.java (hunk 2 above)
Comment (ricklarsen) [inline catalog/spatial/ogc/spatial-ogc-common/src/test/java/org/codice/ddf/spatial/ogc/catalog/common/TestTrustedRemoteSource.java]: yes...

**Comment by ricklarsen** [comment]
@shaundmorris Does it look better now? All the commented out tests should be gone now

**Comment by ryeats** [comment]
:+1:

File: distribution/ddf-common/src/main/resources/etc/system.properties [hunk 4]
```diff
@@ -158,4 +158,5 @@ karaf.secured.services=(&(osgi.command.scope=*)(osgi.command.function=*))
 #org.osgi.framework.security=osgi
 #org.osgi.framework.trust.repositories=${karaf.home}/etc/trustStore.ks
 ws-security.subject.cert.constraints = .*
-
+org.apache.cxf.configuration.jsse.sslAllowedAlgorithms = .*_EXPORT_.*, .*_WITH_AES_.*, .*_WITH_3DES_.*
+org.apache.cxf.configuration.jsse.sslDisallowedAlgorithms = .*_WITH_NULL_.*, .*_DH_anon_.*
```
Comment (stustison) [inline distribution/ddf-common/src/main/resources/etc/system.properties]: This should not be added. We have allowed and not allowed algorithms already. Further, these are the incorrect settings.

File: platform/security/sts/security-sts-ldapclaimshandler/src/main/java/ddf/security/sts/claimsHandler/ClaimsHandlerManager.java [hunk 5]
```diff
@@ -154,24 +147,6 @@ protected LDAPConnectionFactory createLdapConnectionFactory(String url, Boolean
             truststorePass = encryptService.decryptValue(truststorePass);
```
Comment (stustison) [inline platform/security/sts/security-sts-ldapclaimshandler/src/main/java/ddf/security/sts/claimsHandler/ClaimsHandlerManager.java]: Doesn't look like we need to be grabbing these SSL properties anymore if we aren't using them.

File: platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java [hunk 6]
```diff
@@ -412,32 +409,6 @@ protected LDAPConnectionFactory createLdapConnectionFactory(String url, Boolean
 
         LDAPOptions lo = new LDAPOptions();
 
-        try {
```
Comment (stustison) [inline platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java]: Need to manually confirm that startTls and LDAPS still work.

File: distribution/ddf-common/src/main/resources/etc/system.properties (hunk 4 above)
Comment (ricklarsen) [inline distribution/ddf-common/src/main/resources/etc/system.properties]: removed.

File: platform/security/sts/security-sts-ldapclaimshandler/src/main/java/ddf/security/sts/claimsHandler/ClaimsHandlerManager.java (hunk 5 above)
Comment (ricklarsen) [inline platform/security/sts/security-sts-ldapclaimshandler/src/main/java/ddf/security/sts/claimsHandler/ClaimsHandlerManager.java]: removed.

**Comment by djblue** [comment]
:+1:

**Comment by ahoffer** [comment]
`groovy PRBS.groovy PRBS.groovy https://github.com/codice/ddf/pull/253 dib --verbose`

```
09:51:35,579 | ERROR | Event Dispatcher | SolrServerFactory                | e.solr.factory.SolrServerFactory  149 | 381 - catalog-solr-external-provider - 2.9.0.SNAPSHOT | Unable to create HTTP Solr server client (https://localhost:9993/solr/catalog): Expected mime type application/octet-stream but got text/html. <html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"/>
<title>Error 403 Forbidden</title>
</head>
<body><h2>HTTP ERROR 403</h2>
<p>Problem accessing /solr/admin/cores. Reason:
<pre>    Forbidden</pre></p><hr /><i><small>Powered by Jetty://</small></i><br/>  
```


---

Review status: 0 of 24 files reviewed at latest revision, 5 unresolved discussions.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/253)

**Comment by ahoffer** [comment]
HERO PASS. Executed: `groovy PRBS.groovy https://github.com/codice/ddf/pull/253 dib --verbose`
:+1:

**Pull request merged**